script/test

```diff
@@ -219,16 +219,23 @@ do_expect()
         test_patterns=(-name 'cli-*.exp' -o -name 'simulation-*.exp')
     fi
 
+    local temp_dir
+    temp_dir=$(mktemp -d)
+    trap 'rm -r "$temp_dir"' EXIT
+
+    local log_file
+    log_file="${temp_dir}/log_expect"
     while read -r script; do
         sudo rm -rf tmp
         {
             if [[ ${OT_NATIVE_IP} == 1 ]]; then
-                OT_COMMAND="${ot_command}" RCP_COMMAND="${rcp_command}" sudo -E "${script}"
+                OT_COMMAND="${ot_command}" RCP_COMMAND="${rcp_command}" sudo -E expect -df "${script}" 2>"${log_file}"
```

Comment:
`test` script has the `VERBOSE` option. 
What about we print the debug log when `VERBOSE=1` and then use `VERBOSE=1` in GitHub actions.

Reply:
Changed to always print debug logs when `VERBOSE=1`.
But didn't turn on verbose mode in GitHub actions. That will cause too many outputs. We only need these logs when tests fail.

Reply:
If we just use `VERBOSE=1` in github actions, we don't need all these logic of redirecting log to a tmp file and printing it when failed.
Besides, having more debug logs in github action does not seems to be an issue for me.

Reply:
The concern about the huge number of logs is that in GitHub Action when it fails we have to scroll to the bottom manually to see what happened, the loading and scrolling lags with too many logs.
Also when running locally, I think it's nice to have detailed logs printed for failed cases by default, so we don't have to run again with debug mode if something fails.
Another benefit of redirecting is that `stdout` and `stderr` won't interleave, which makes them easier to read.

Reply:
> The concern about the huge number of logs is that in GitHub Action when it fails we have to scroll to the bottom manually to see what happened, the loading and scrolling lags with too many logs.

This can be solved by using `View raw logs` or `Download log archive`. BTW, building openthread itself already generates massive logs.

> Also when running locally, I think it's nice to have detailed logs printed for failed cases by default, so we don't have to run again with debug mode if something fails.

It's nice to always print logs if failed but it's also very simple to run it again locally. 
I am fine with the current strategy but I do prefer a simpler solution which has no obvious disadvantages.

Reply:
@simonlingoogle The certification tests only print logs of failed cases. I don't think it's necessary to print passed tests. Too many unnecessary logs increases the difficulty to debug when bad things happen.